python/src/kagent/models/vertexai/_anthropic_vertex_client.py

```diff
@@ -0,0 +1,121 @@
+import inspect
+
+from anthropic import AsyncAnthropicVertex
+from autogen_core import Component
+from autogen_ext.models.anthropic import BaseAnthropicChatCompletionClient
+from google.auth import load_credentials_from_dict
+from typing_extensions import Any, Dict, Mapping, Optional, Self, Set, Unpack
+
+from .config import AnthropicVertexAIClientConfiguration, ModelInfo
+
+# Common parameters for message creation
+anthropic_message_params = {
+    "system",
+    "messages",
+    "max_tokens",
+    "temperature",
+    "top_p",
+    "top_k",
+    "stop_sequences",
+    "tools",
+    "tool_choice",
+    "stream",
+    "metadata",
+}
+disallowed_create_args = {"stream", "messages"}
+required_create_args: Set[str] = {"model"}
+
+anthropic_init_kwargs = set(inspect.getfullargspec(AsyncAnthropicVertex.__init__).kwonlyargs)
+
+
+def _create_args_from_config(config: Mapping[str, Any]) -> Dict[str, Any]:
+    create_args = {k: v for k, v in config.items() if k in anthropic_message_params or k == "model"}
+    create_args_keys = set(create_args.keys())
+
+    if not required_create_args.issubset(create_args_keys):
+        raise ValueError(f"Required create args are missing: {required_create_args - create_args_keys}")
+
+    if disallowed_create_args.intersection(create_args_keys):
+        raise ValueError(f"Disallowed create args are present: {disallowed_create_args.intersection(create_args_keys)}")
+
+    return create_args
+
+
+def _anthropic_client_from_config(config: Mapping[str, Any]) -> AsyncAnthropicVertex:
+    # Filter config to only include valid parameters
+    client_config = {k: v for k, v in config.items() if k in anthropic_init_kwargs}
+    return AsyncAnthropicVertex(**client_config)
+
+
+class AnthropicVertexAIChatCompletionClient(
+    BaseAnthropicChatCompletionClient, Component[AnthropicVertexAIClientConfiguration]
+):
+    component_type = "model"
+    component_config_schema = AnthropicVertexAIClientConfiguration
+    component_provider_override = "kagent.models.vertexai.AnthropicVertexAIChatCompletionClient"
+
+    def __init__(self, **kwargs: Unpack[AnthropicVertexAIClientConfiguration]):
```

Comment:
Why not just pass in the whole config object, or have typed args?

Reply:
with config object the caller has to pass in an object vs. if it's using kwargs you only pass in the args you need/want (you still get type hints and all that)